src/frontend/js/modules/session.spec.js

```diff
@@ -366,4 +366,66 @@ describe('The esn.session Angular module', function() {
 
   });
 
+  describe('the sessionInitESNController', () => {
+    let userAPI, esnAuth, self;
+
+    beforeEach(function() {
+      self = this;
+
+      userAPI = {
+        currentUser: sinon.stub().returns($q.when({})),
+        user: sinon.stub().returns($q.when({}))
+      };
+
+      esnAuth = {
+        signInCompletePromise: {
+          then: callback => callback()
+        },
+        signout: sinon.spy(),
+        init: sinon.stub().returns($q.when({}))
+      };
+
+      angular.mock.module(function($provide) {
+        $provide.value('userAPI', userAPI);
+        $provide.value('esnAuth', esnAuth);
+      });
+
+      angular.mock.inject(function($rootScope, $controller, session) {
+        this.$scope = $rootScope.$new();
+        this.$rootScope = $rootScope;
+        this.$controller = $controller;
+        this.session = session;
+      });
+    });
+
+    function initController() {
+      const controller = self.$controller('sessionInitESNController', {
+        $rootScope: self.$rootScope,
+        $scope: self.$scope
+      });
+
+      return controller;
+    }
+
+    it('should signout the user if it is unauthorized ( broken session )', function() {
+      esnAuth.init = sinon.stub().returns($q.reject({
+        code: 401
+      }));
+
+      initController();
+      this.$scope.$digest();
+
+      expect(esnAuth.signout).to.have.been.called;
+    });
+
+    it('shouldn\'t signout the user when a non 401 error was received', function() {
+      esnAuth.init = sinon.stub().returns($q.reject({
+        code: 500
+      }));
+
+      initController();
+
+      expect(esnAuth.signout).to.not.have.been.called;
```

Comment:
You are missing `this.$scope.$digest();` so the test is always success

```suggestion
      initController();
       this.$scope.$digest();

      expect(esnAuth.signout).to.not.have.been.called;
```

Reply:
calling `this.$scope.$digest();` will fail with `[$rootScope:inprog] $digest already in progress`
i called $timeout to safely do an $apply()

Reply:
without `digest`, did you try to check 

```
 expect(esnAuth.signout).to.have.been.called;
```

I guess it works too.

Reply:
I'm checking if it's not called.